pyoanda/order.py

```diff
@@ -55,9 +56,12 @@ def check(self):
                 type(self.expiry), datetime)
             raise TypeError(msg)
 
-        if hasattr(self, "price") and not isinstance(self.price, (int, float)):
-            msg = "Expiry must be int or float, '{0}' found".format(
-                type(self.price))
-            raise TypeError(msg)
+        if hasattr(self, "price"):
+            try:
+                Decimal(self.price)                
```

Comment:
I think try: float(self.price) except: ValueError is probably sufficient, you don't need to pull in decimal to do the check.